src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -29,25 +39,123 @@ export default class FileField extends React.Component {
     }
   }
 
-  onAddFile = (event, index = null) => {
+  componentDidMount() {
+    // The File object is not preserved in the save-in-progress data
+    // We need to remove these entries; an empty `file` is included in the
+    // entry, but if API File Object still exists (within the same session), we
+    // can't use Object.keys() on it because it returns an empty array
+    const formData = (this.props.formData || []).filter(
+      // keep - file may not exist (already uploaded)
+      // keep - file may contain File object; ensure name isn't empty
+      // remove - file may be an empty object
+      data => !data.file || (data.file?.name || '') !== '',
+    );
+    if (formData.length !== (this.props.formData || []).length) {
+      this.props.onChange(formData);
+    }
+  }
+
+  focusAddAnotherButton = () => {
+    // focus on span pseudo-button, not the label
+    focusElement(`#${this.props.idSchema.$id}_add_label span`);
+  };
+
+  onSubmitPassword = (file, index, password) => {
+    if (file && password) {
+      this.onAddFile({ target: { files: [file] } }, index, password);
+    }
+  };
+
+  isFileEncrypted = async (file, index) => {
+    const { onChange, formData } = this.props;
+    return checkForEncryptedPdf(file)
+      .then(isEncrypted => {
+        if (isEncrypted) {
+          onChange(
+            _.set(
+              index,
+              { file, name: file.name, index, isEncrypted: true },
+              formData || [],
+            ),
+          );
+        }
```

Comment:
It seems like lodash can be avoided here:


```js
    const { onChange, formData = [] } = this.props;
    return checkForEncryptedPdf(file)
      .then(isEncrypted => {
        if (isEncrypted) {
          onChange(() => {
            formData[index] = {
              file,
              name: file.name,
              index,
              isEncrypted: true,
            };
          });
 
```

Reply:
Oh good idea! I was copying how it was done elsewhere in the code. I guess updating them all would be out of scope...